main/core/messagebroker.ts

```diff
@@ -99,6 +103,39 @@ export class MessageBroker<T = any> implements IMessageBroker<T> {
         delete this.channelLookup[channelName];
     }
 
+    /**
+     * Register an adapter with the message broker
+     * @param adapter The adapter to register
+     */
+    public registerAdapter(adapter: IMessageBrokerAdapter<T>): void {
```

Comment:
I think it would be better if the messageBroker assigned an id and returned it from here. Either that or you pass an id into this function but then you might get conflicts. If the adaptor assigns the id then there is no chance of a conflict.